### PR TITLE
drivers/tsl2561: apply unified params definition scheme + cleanup

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR update the params definitions scheme for the tsl2561 device driver.

### Issues/PRs references

Initially done in #7937 and related to #7519

#6577

### CODE REVIEW

File: sys/auto_init/saul/auto_init_tsl2561.c [hunk 1]
```diff
@@ -28,42 +28,44 @@
 /**
  * @brief   Define the number of configured sensors
  */
-#define TSL2561_NUMOF    (sizeof(tsl2561_params) / sizeof(tsl2561_params[0]))
+#define TSL2561_NUM      (sizeof(tsl2561_params) / sizeof(tsl2561_params[0]))
```
Comment (smlng) [inline sys/auto_init/saul/auto_init_tsl2561.c]: why this renaming? Its `I2C_NUMOF`, too?

Comment (aabadie) [reply sys/auto_init/saul/auto_init_tsl2561.c]: Fixed

**Comment by MichelRottleuthner** [comment]
tested on pba-d-01-kw2x, works!

**Comment by smlng** [comment]
@MichelRottleuthner care to approve and merge then?

**Comment by MichelRottleuthner** [comment]
@aabadie do you want to squash the param/i2c changes into separate commits?

**Comment by aabadie** [comment]
@MichelRottleuthner, for me the history of commits is ok in this PR. It can be merged as is.

**Pull request merged**